features/support/files_helpers.rb

```diff
@@ -13,30 +13,17 @@ def files_in branch:
 # Returns a table of all files in all branches.
 # This is for comparing against expected files in a Cucumber table.
 def files_in_branches
-  existing_local_branches.map do |branch|
+  result = [%w(BRANCH NAME CONTENT)]
+  existing_local_branches(order: :main_first).each do |branch|
     files_in(branch: branch).map do |file|
       content = content_of file: file, for_sha: branch
-      { branch: branch, name: file, content: content }
+      result << [branch, file, content]
     end
-  end.flatten
+  end
+  result
```

Comment:
Yeah, but this is more performant and in my opinion readable, since it only creates a single array now and returns that. The previous version created a sizeable nested array, and then flattened that into another one.